src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -1315,14 +1348,83 @@ void Graph::InferDynamic(SyncInferRequest* request) {
 }
 
 inline void Graph::ExecuteNode(const NodePtr& node, const dnnl::stream& stream) const {
-    DUMP(node, getConfig().debugCaps, infer_count);
-
-    OV_ITT_SCOPED_TASK(itt::domains::intel_cpu, node->profiling.execute);
-    DEBUG_LOG(*node);
-    if (node->isDynamicNode()) {
-        node->executeDynamic(stream);
+    if (!node->parallelWith.empty()) {
+        // run nodes in parallel
+        auto& parallelNodes = node->parallelWith;
+        if (node == parallelNodes[0]) {
+            auto cpuExecutor =
+                std::dynamic_pointer_cast<ov::threading::CPUStreamsExecutor>(context->getStreamExecutor());
```

Comment:
> Since the context is an entity specific for the cpu plugin, why can't we simply store the `CPUStreamsExecutor` there, instead of casting every time? Are there any different executors which are allowed to be set to the context?

Sure. `CPUStreamsExecutor` can be achieved in graphcontext construtor. I add a new member function.

Reply:
If we store a specific type, I think we don't need an accessor returning an abstract type, it can be easily upcasted by the user. Could you please remove `ov::threading::IStreamsExecutor::Ptr streamExecutor;` as it's redundant?

Reply:
> If we store a specific type, I think we don't need to store the same pointer two times, it can be easily upcasted by the user. Could you please remove `ov::threading::IStreamsExecutor::Ptr streamExecutor;` and its accessor as they are redundant?

sure, removed. specific type can meet our needs.